### PR TITLE
Initial Commit (ESRGAN)

### PR DESCRIPTION
**Pull request by captain-pool**
All commits on ESRGAN will be on this branch.
Feel Free to comment.
CC: @srjoglekar246 @vbardiovskyg

### CODE REVIEW

**Comment by srjoglekar246** [comment]
Why a separate branch? Is this code rough for now? 

Otherwise, we could just have a new folder for ESRGAN functionality, with PRs being sent to it for each incremental change - this way, we won't have to review O(1000) lines of code at the same time :-).Each of your commits would qualify for a separate PR, for example.

**Comment by captain-pool** [comment]
> Why a separate branch? Is this code rough for now?
> 
> Otherwise, we could just have a new folder for ESRGAN functionality, with PRs being sent to it for each incremental change - this way, we won't have to review O(1000) lines of code at the same time :-).Each of your commits would qualify for a separate PR, for example.

Yep, as of now the code is rough. None of the code written here is tested. So, I'm kind of skeptic of committing directly to the master branch. I'll merge with all the basic functionalities tested, like syntax errors and all. Later on, I'm planning to create PR for every functionality that will be added. What do you think?

**Comment by srjoglekar246** [comment]
Thats seems okay, as long as all the code gets merged into Master & sent in as separate per-component PRs.

**Comment by captain-pool** [comment]
Hey @srjoglekar246 I merged it by mistake.
This PR Adds the Generator and Loss Functions for ESRGAN. The model builds perfectly, and the output gets scaled up by the exact amount it is needed to be.
I've added a settings classes which is a singleton class, which is being called around in various parts of the code, and the hyperparams are fetched from a config.yaml file.
In `main.py` one needs to create an object of `settings()` passing the `filename` only once. later in various parts the properties from that files are used.

Though the code runs, in the exact way, can you take a look and suggest some changes which I can fix in the next PRs?
